Source/engine/render/scrollrt.cpp

```diff
@@ -867,7 +867,7 @@ void DrawDungeon(const Surface &out, Point tilePosition, Point targetBufferPosit
 		DrawDeadPlayer(out, tilePosition, targetBufferPosition);
 	}
 	int8_t playerId = dPlayer[tilePosition.x][tilePosition.y];
-	if (playerId > 0 && playerId <= MAX_PLRS) {
+	if (static_cast<size_t>(playerId - 1) <= Players.size()) {

```

Comment:
This will start to render the player at both the leaving and entering position. Only positive numbers should be accepted here.

Reply:
This does accept only the positive numbers. static_cast<size_t>(-1) is a huge number.

However, the condition should be:

```diff
-	if (static_cast<size_t>(playerId - 1) <= Players.size()) {
+	if (static_cast<size_t>(playerId - 1) < Players.size()) {
```